### PR TITLE
fix: prevent crash when git is not avail

### LINKED ISSUE #2: Curious if you test this on a non-git folder?
I'm just wondering if we should add error-handling or a try/catch wrapper.

### PR DESCRIPTION
**Pull request by pke**
The appInfo will then just not contain the `hash` and `tag` members.

Closes #2

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -2,7 +2,24 @@ const readPkgUp = require('read-pkg-up');
...
+ * A safe way to query for git infos.
+ *
+ * @return {object|undefined} Returns {hash, gitTag} or undefined in case of an error
+ */
+async function getLastCommit() {
+  try {
+    const lastCommitLog = new LastCommitLog();
+    const { hash, gitTag } = await lastCommitLog.getLastCommit();
+    return { hash, gitTag };
+  } catch (err) {}
```
Comment (niftylettuce) [inline index.js:22]: Can you change it so we can pass a custom logger?

```diff
-async function parseAppInfo() {
+async function parseAppInfo(logger = console) {
```

Then make it so `getLastCommit` accepts this logger as well?

```diff
-async function getLastCommit() {
+async function getLastCommit(logger) {
  try {
    const lastCommitLog = new LastCommitLog();
    const { hash, gitTag } = await lastCommitLog.getLastCommit();
    return { hash, gitTag };
-  } catch (err) {}
+  } catch (err) {
+    logger.error(err);
+    return {};
  }
}
```

File: index.js [hunk 2]
```diff
@@ -4,14 +4,19 @@ const LastCommitLog = require('last-commit-log');
...
  */
-async function getLastCommit() {
+async function getLastCommit(logger) {
   try {
     const lastCommitLog = new LastCommitLog();
     const { hash, gitTag } = await lastCommitLog.getLastCommit();
     return { hash, gitTag };
-  } catch (err) {}
+  } catch (err) {
+    if (logger) {
```
Comment (niftylettuce) [inline index.js]: On second thought, I'd suggest to just remove this commit.  Users that don't use git probably don't want flooded logger messages.

Comment (niftylettuce) [reply index.js]: Have you tested with and without git tag?  We shouldn't add a property git tag if it's missing or blank.

Comment (pke) [reply index.js]: Will add that later.

Comment (niftylettuce) [reply index.js]: Thanks again for your efforts, this is incredibly resourceful at a time of need.

**Comment by niftylettuce** [comment]
@pke can you fix the tests and/or lower code coverage threshold?

looks good though 👍

**Comment by pke** [comment]
Unfortunately 100% coverage with the git calls isn't possible. 
But anyway it's good enough. I've tested it by removing the git folder. And tag was empty when I tested with git folder back in place.

**Pull request merged**